modules/clients/src/test/java/org/apache/ignite/jdbc/JdbcAuthorizationTest.java

```diff
@@ -0,0 +1,440 @@
+    }
+
+    /**
+     * Tests CREATE TABLE query cache permissions check.
+     */
+    @Test
+    public void testCreateTableCachePermissions() throws Exception {
+        StatementProvider stmt = of("CREATE TABLE cache_perm_table_create(id LONG PRIMARY KEY, val varchar) WITH " +
+            "\"TEMPLATE=REPLICATED, CACHE_NAME=" + TEST_CREATE_TABLE_CACHE + "\";");
+
+        assertAuthorizationFailed(stmt, EMPTY_PERMS_USER);
+
+        execute(stmt, CACHE_CREATE_CACHE_PERMS_USER);
+    }
+
+    /**
+     * Tests DROP TABLE query system permissions check.
+     */
+    @Test
+    public void testDropTableSystemPermissions() throws Exception {
+        String table = "test_sys_perm_table_drop";
+
+        createTable(table);
+
+        StatementProvider stmt = of("DROP TABLE " + table + ';');
+
+        assertAuthorizationFailed(stmt, EMPTY_PERMS_USER);
+
+        execute(stmt, CACHE_DESTROY_SYS_PERMS_USER);
+    }
+
+    /**
+     * Tests DROP TABLE query cache permissions check.
+     */
+    @Test
+    public void testDropTableCachePermissions() throws Exception {
+        String table = "test_cache_perm_table_drop";
+
+        execute(of("CREATE TABLE " + table + "(id LONG PRIMARY KEY, str_col varchar, long_col LONG)" +
+                " WITH \"TEMPLATE=REPLICATED, CACHE_NAME=" + TEST_DROP_TABLE_CACHE + "\";"),
+            CACHE_CREATE_SYS_PERM_USER);
+
+        StatementProvider stmt = of("DROP TABLE " + table + ';');
+
+        assertAuthorizationFailed(stmt, EMPTY_PERMS_USER);
+
+        execute(stmt, CACHE_DESTROY_CACHE_PERMS_USER);
+    }
+
+    /**
+     * Tests ALTER TABLE query permissions check.
+     */
+    @Test
+    public void testAlterTableAddColumn() throws Exception {
+        String table = "test_table_add_column";
+
+        createTable(table);
+
+        execute(of("ALTER TABLE " + table + " ADD test_add_column LONG;"), EMPTY_PERMS_USER);
+    }
+
+    /**
+     * Tests ALTER TABLE DROP COLUMN query permissions check.
+     */
+    @Test
+    public void testAlterTableDropColumn() throws Exception {
+        String table = "test_table_drop_column";
+
+        createTable(table);
+
+        execute(of("ALTER TABLE " + table + " DROP COLUMN long_col;"), EMPTY_PERMS_USER);
+    }
+
+    /**
+     * Tests CREATE INDEX queries permissions check.
+     */
+    @Test
+    public void testCreateIndex() throws Exception {
+        String table = "test_table_create_index";
+
+        createTable(table);
+
+        execute(of("CREATE INDEX test_create_idx ON " + table + "(id ASC);"), EMPTY_PERMS_USER);
+    }
+
+    /**
+     * Tests DROP INDEX queries permissions check.
+     */
+    @Test
+    public void testDropIndex() throws Exception {
+        String table = "test_table_drop_index";
+
+        createTable(table);
+
+        execute(of("CREATE INDEX test_drop_idx ON " + table + "(id ASC);"), EMPTY_PERMS_USER);
+
+        execute(of("DROP INDEX test_drop_idx ON " + table + ';'), EMPTY_PERMS_USER);
```

Comment:
This test do the same as `testCreateIndex` + one extra action. In this case, `testCreateIndex` method is redundant and can be removed (this method should be renamed to `testCreateAndDropIndex`).

Also, what do we check in these tests? It looks strange that we can't limit creating and dropping columns and indexes and check it like it's expected behavior. A test should assert some reasonable statement, I see no statement to assert here.

Reply:
I agree that it looks strange. I have asked the corresponding question on the dev-list but there was no answer yet. Could we keep this tests to only show the current authorization behavior?

Reply:
I think it's better to avoid checking unexpected results in tests even to show current behavior.

Reply:
I fixed comments mentioned by you. Could you please take a look?